samcli/lib/telemetry/metric.py

```diff
@@ -124,21 +124,28 @@ def hello_command():
 
     """
 
+    @wraps(func)
     def wrapped(*args, **kwargs):
-        telemetry = Telemetry()
-
         exception = None
         return_value = None
         exit_reason = "success"
         exit_code = 0
 
         duration_fn = _timer()
+
+        ctx = None
+        try:
+            ctx = Context.get_current_context()
+        except RuntimeError:
+            LOG.debug("Unable to find Click Context for getting session_id.")
+
         try:
+            if ctx and ctx.exception:
+                raise ctx.exception
```

Comment:
The reason that this is happening here (and instead of somewhere else in some other control flow) is that an exception that got saved in click would have been raised before the CLI method. We'd want to stop execution there and not actually run the command.

This try block handles `UserException` and `Exception` already, so it would end up catching anything we stuff into `ctx.exception`. Raising it within the try just lets use reuse that existing behaviour